ospfd/ospf_vty.c

```diff
@@ -377,65 +377,142 @@ DEFPY (no_ospf_router_id,
 }
 
 
-static void ospf_passive_interface_default(struct ospf *ospf, uint8_t newval)
+static void ospf_passive_interface_set(struct ospf *ospf,
+				       struct interface *ifp, int create)
 {
-	struct vrf *vrf = vrf_lookup_by_id(ospf->vrf_id);
-	struct listnode *ln;
-	struct interface *ifp;
+	struct listnode *cnode;
+	struct connected *co;
+	struct route_node *rn;
 	struct ospf_interface *oi;
+	struct in_addr area_id = {0};
+	struct prefix_ipv4 ipv4, *ip_ptr;
+
+	if (create == OSPF_IF_PASSIVE) {
+		/* create passive-interface from connected address */
+		ospf->passive_interface_default = OSPF_IF_PASSIVE;
```

Comment:
This doesn't make sense. You're setting `ospf->passive_interface_default` to `OSPF_IF_PASSIVE` here and setting it to `OSPF_IF_ACTIVE` a few lines below. The problem is that a) no one is checking the value of this variable and b) this function acts on interfaces but `passive_interface_default` is tied to OSPF areas and not OSPF interfaces.